### PR TITLE
[DOCS] Enables testing for monitoring examples

### PR DESCRIPTION
**Pull request by lcawl**
This PR attempts to enable code snippet testing for the monitoring/indices.asciidoc page, which is externalized here: 
https://www.elastic.co/guide/en/elasticsearch/reference/master/config-monitoring-indices.html

The first example needs to be skipped, however, since it results in the following 404 error otherwise:

> 1> [2018-06-05T19:46:44,113][INFO ][o.e.s.XDocsClientYamlTestSuiteIT] [test {yaml=en/monitoring/indices/line_11}]: after test
  1> [2018-06-05T19:46:44,113][ERROR][o.e.s.XDocsClientYamlTestSuiteIT] This failing test was generated by documentation starting at en/monitoring/indices.asciidoc:line_11. It may include many snippets. See Elasticsearch's docs/README.asciidoc for an explanation of test generation.
  2> REPRODUCE WITH: ./gradlew :x-pack:docs:integTestRunner -Dtests.seed=CF22EFB6F9816360 -Dtests.class=org.elasticsearch.smoketest.XDocsClientYamlTestSuiteIT -Dtests.method="test {yaml=en/monitoring/indices/line_11}" -Dtests.security.manager=true -Dtests.locale=be -Dtests.timezone=CNT
FAILURE 0.40s | XDocsClientYamlTestSuiteIT.test {yaml=en/monitoring/indices/line_11} <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: Failure at [en/monitoring/indices:10]: expected [2xx] status code but api [raw[method=GET path=_template/.monitoring-*]] returned [404 Not Found] [{}]
   >    at __randomizedtesting.SeedInfo.seed([CF22EFB6F9816360:4776D06C577D0E98]:0)
   >    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:373)
   >    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:350)
   >    at jdk.internal.reflect.GeneratedMethodAccessor11.invoke(Unknown Source)
   >    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   >    at java.base/java.lang.reflect.Method.invoke(Method.java:564)
   >    at java.base/java.lang.Thread.run(Thread.java:844)
   > Caused by: java.lang.AssertionError: expected [2xx] status code but api [raw[method=GET path=_template/.monitoring-*]] returned [404 Not Found] [{}]

My guess is that this error occurs because the test occurs before any templates have been created. If there are commands that could force that to occur, we could use those in the test setup.

### CODE REVIEW

File: x-pack/docs/en/monitoring/indices.asciidoc [hunk 1]
```diff
@@ -11,6 +11,8 @@ You can retrieve the templates through the `_template` API:
 ----------------------------------
 GET /_template/.monitoring-*
 ----------------------------------
+// CONSOLE
+// TEST[skip:404 error]
```
Comment (nik9000) [inline x-pack/docs/en/monitoring/indices.asciidoc]: Will `// TEST[catch:missing]` do it?

Comment (nik9000) [reply x-pack/docs/en/monitoring/indices.asciidoc]: Or we could make sure that the templates are created first.

Comment (lcawl) [reply x-pack/docs/en/monitoring/indices.asciidoc]: Thanks @nik9000, catch:missing worked.  Can we document or link to that list of keywords somewhere?

Comment (nik9000) [reply x-pack/docs/en/monitoring/indices.asciidoc]: https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/test/README.asciidoc#required-operators has a list that should be up to date. I tend to look [here](https://github.com/elastic/elasticsearch/blob/master/test/framework/src/main/java/org/elasticsearch/test/rest/yaml/section/DoSection.java#L336-L351) though.

Comment (nik9000) [reply x-pack/docs/en/monitoring/indices.asciidoc]: We could probably do with a link from https://github.com/elastic/elasticsearch/blob/master/docs/README.asciidoc though.

**Review by nik9000 — APPROVED** [review]

**Pull request merged**